pywatts/modules/postprocessing/merger.py

```diff
@@ -0,0 +1,95 @@
+    current time-index.
+
+    :param name: The name of the module
+    :type name: str
+    :param method: The method indicates the merging method. Method can be 'mean', 'median', or an integer between 0 and
+                   -1.
+                   For each time step, the 'mean' method takes all values of the input time series that corresponds
+                   to the same time, and calculates the mean.
+                    For each time step, the 'median' method takes all values of the input time series that corresponds
+                   to the same time, and calculates the median.
+                   If the method is an integer k, the k-th value in the second dimension is taken for each time step.
+                   If k is negative, the k-th values in the second dimension before the last is selected.
+                   If the absolute value of k is greater than the size of the horizon dimension, then the value is
+                   clipped.
+    :type method: Union[str,int]
+    :raises WrongParameterException: If method is not an integer, 'mean' or 'median'
+    """
+    def __init__(self, name: str = "merger", method: Union[str, int] = "mean"):
+        super().__init__(name)
+        self._check_and_set_method(method)
+
+    def _check_and_set_method(self, method):
+        if method in ["mean", "median"] or isinstance(method, int):
+            self.method = method
+        else:
+            raise WrongParameterException("The parameter method has an invalid value",
+                                          "Try one of the following strings: 'mean' 'median",
+                                          self)
+
+    def get_params(self) -> Dict[str, object]:
+        """
+        Get parameters for the Merger module.
+        :return: Parameters as dict.
+        :rtype: Dict[str, object]
+        """
+        return {
+            "method" : self.method
+        }
+
+    def set_params(self, method: Union[str, int] = None):
+        """
+        Set parameters for the Merger module.
+        :param method:
+        :type method: Union[str,int]
+        :raises WrongParameterException: If method is not an integer, 'mean' or 'median'
+        """
+        if method is not None:
+            self._check_and_set_method(method)
+
+    def transform(self, x: xr.DataArray) -> xr.DataArray:
+        """
+        Merge the given time series by mean, median or selecting one specific value.
+        :return: DataArray of the resulting merged time series.
+        :rtype: xr.DataArray
+        """
+        horizon = x.values.shape[-1]
+        if self.method == "mean":
+            r = []
+            for i in range(horizon):
+                r.append(np.concatenate(
+                    [np.full(fill_value=np.nan, shape=(horizon - 1 - i)),
+                     x.values[:, i],
+                     np.full(fill_value=np.nan, shape=(i,))]))
```

Comment:
You might want to extract this part as a method since the same code is also used for the median in lines 82-87.

Reply:
Thank you. I will extract the code